cashctrl_api/client.py

```diff
@@ -212,39 +212,71 @@ def flatten_nodes(nodes, parent_path=''):
             # Remove first node (the system root) from paths
             df['path'] = df['path'].str.replace('^/+[^/]+', '', regex=True)
 
         return df.sort_values('path')
 
-    def update_categories(self, resource: str, target: List[str],
-                        delete: bool = False):
+    def update_categories(self, resource: str, target: List[str] | dict,
+                          delete: bool = False):
         """
         Updates the server's category tree for a specified resource,
         synchronizing it with the provided category list.
 
         Args:
             resource (str): Resource type ('account', 'file', etc.).
-            target (List[str]): Target category paths in Unix-like format.
+            target (List[str] | dict): Target category paths in Unix-like format.
             delete (bool, optional): If True, deletes categories not present
                                      in the provided list. Defaults to False.
         """
+        if resource == 'account' and not isinstance(target, dict):
+            raise ValueError('Accounts target should be a dict of groups and associated account numbers')
+        elif resource != 'account' and isinstance(target, dict):
+            raise ValueError('Target categories should be a list for this resource')
+
         category_list = self.list_categories(resource)
         categories = dict(zip(category_list['path'], category_list['id']))
 
         if delete:
-            target_df = pd.Series(pd.Series(target).unique())
-            to_delete = [node for node in category_list['path']
-                        if not target_df.str.startswith(node).any()]
+            if isinstance(target, list):
+                target_series = pd.Series(target)
+            else:
+                target_series = pd.Series(target).index
+            target_df = pd.Series(target_series.unique())
+            to_delete = [node for node in category_list['path']if not target_df.str.startswith(node).any()]
+
             if to_delete:
                 to_delete.sort(reverse=True) # Delete from leaf to root
                 delete_ids = [str(categories.pop(path)) for path in to_delete]
                 self.post(f"{resource}/category/delete.json",
                           params={'ids': ','.join(delete_ids)})
 
+        # Update account category number if target differs from remote
+        if resource == 'account':
+            dict_df = pd.DataFrame(list(target.items()), columns=['path', 'new_number'])
+            merged = category_list.merge(dict_df, on='path', how='inner')
+            merged.dropna(subset=['number', 'new_number'], inplace=True)
+            update = merged[merged['number'] != merged['new_number']]
+            for row in update.to_dict('records'):
+                params = {
+                    'id': row['id'],
+                    'name': row['path'],
+                    'number': row['new_number'],
+                    'parentId': row['parentId']
```

Comment:
Do we actually need to pass 'name' and 'parentId'?
Many REST APIs accept if we just pass the modified fields, and only passing the modified fields enhances code clarity.

Reply:
Yeah, but not in this case

I tried to send such a request:
```python
                    params = {
                        'id': row['id'],
                        'number': target[row['path']],
                    }
```
and get this in result:
```python
FAILED tests/test_categories.py::test_account_category_update - requests.exceptions.RequestException: API call failed. number: The value of this field must be a number. / name: This field cannot be empty. / parentId: This field cannot be empty.
```

Reply:
Thanks!